snapcraft/internal/build_providers/_lxd/_lxd.py

```diff
@@ -159,29 +158,28 @@ def __init__(
     def _run(
         self, command: Sequence[str], hide_output: bool = False
     ) -> Optional[bytes]:
-        self._ensure_container_running()
+        # Sanity check - developer error if container not initialized.
+        if self._container is None:
+            raise RuntimeError("Attempted to use container before starting.")
 
-        env_command = super()._get_env_command()
+        self._ensure_container_running()
 
-        # TODO: use pylxd
-        cmd = [self._LXC_BIN, "exec", self.instance_name, "--"]
-        cmd.extend(env_command)
+        cmd = list(super()._get_env_command())
         cmd.extend(command)
-        self._log_run(cmd)
+        logger.debug(f"Executing in {self.instance_name}: {cmd}")
+        (exit_code, stdout, stderr) = self._container.execute(cmd)
 
-        output = None
-        try:
-            if hide_output:
-                output = subprocess.check_output(cmd)
-            else:
-                subprocess.check_call(cmd)
-        except subprocess.CalledProcessError as process_error:
+        output = stdout + "\n\nSTDERR: " + stderr
```

Comment:
This should be streamed and piped out to stdout and stderr as they occur. It may require extending pylxd to support proper websockets for this and the reason we just went with lxc exec.

Reply:
docstring says this needs to be a callable

Reply:
Can you advise on how to make this a callable?